### PR TITLE
s390x: fail on error during reading of SecureExecution sysfs flag

### PR DESCRIPTION
**Pull request by nikita-dubrovskii**

### CODE REVIEW

File: src/libostree/ostree-bootloader-zipl.c [hunk 1]
```diff
@@ -110,11 +110,16 @@ _ostree_bootloader_zipl_write_config (OstreeBootloader  *bootloader,
...
+                                                     GError **error)
+{
+  if (access (SECURE_EXECUTION_SYSFS_FLAG, F_OK) != 0)
+    return FALSE;
   gsize len = 0;
-  g_autofree char *data = glnx_file_get_contents_utf8_at (-1, SECURE_EXECUTION_SYSFS_FLAG, &len, cancellable, NULL);
+  g_autofree char *data = glnx_file_get_contents_utf8_at (-1, SECURE_EXECUTION_SYSFS_FLAG, &len, cancellable, error);
   if (!data)
-    return FALSE;
+    return -1;
```
[the remark this answers is not in the document]
Comment (cgwalters) [reply src/libostree/ostree-bootloader-zipl.c:122]: There's already `ot_openat_ignore_enoent()` for this.

**Comment by cgwalters** [comment]
Can you try this?
```
diff --git a/src/libostree/ostree-bootloader-zipl.c b/src/libostree/ostree-bootloader-zipl.c
index b1aa03dc..c570171a 100644
--- a/src/libostree/ostree-bootloader-zipl.c
+++ b/src/libostree/ostree-bootloader-zipl.c
@@ -111,16 +111,19 @@ _ostree_bootloader_zipl_write_config (OstreeBootloader  *bootloader,
 }
 
 // On failure returns -1, otherwise TRUE or FALSE
-static gint _ostree_secure_execution_is_enabled (GCancellable *cancellable,
-                                                     GError **error)
+static gboolean
+_ostree_secure_execution_is_enabled (gboolean *out_enabled,
+                                     GCancellable *cancellable,
+                                     GError **error)
 {
-  if (access (SECURE_EXECUTION_SYSFS_FLAG, F_OK) != 0)
+  glnx_fd_close int fd = -1;
+  if (!ot_openat_ignore_enoent (AT_FDCWD, SECURE_EXECUTION_SYSFS_FLAG, &fd, error))
     return FALSE;
-  gsize len = 0;
-  g_autofree char *data = glnx_file_get_contents_utf8_at (-1, SECURE_EXECUTION_SYSFS_FLAG, &len, cancellable, error);
+  g_autofree char *data = glnx_fd_readall_utf8 (fd, NULL, cancellable, error);
   if (!data)
-    return -1;
-  return strstr (data, "1") != NULL;
+    return FALSE;
+  *out_enabled = strstr (data, "1") != NULL;
+  return TRUE;
 }
 
 static gboolean
@@ -343,8 +346,8 @@ _ostree_bootloader_zipl_post_bls_sync (OstreeBootloader  *bootloader,
     return TRUE;
 
   /* Try with Secure Execution */
-  int se = _ostree_secure_execution_is_enabled (cancellable, error);
-  if (se == -1)
+  gboolean se = FALSE;
+  if (!_ostree_secure_execution_is_enabled (&se, cancellable, error))
     return FALSE;
   if (se)
     {
```

**Comment by cgwalters** [comment]
The rules around `GError` are honestly obtuse and confusing - they *do* make sense once you've learned them at least.

Ultimately of course this is basically a manual replication of what in Rust would be `fn se_is_enabled() -> Result<bool>` where the compiler would check everything is correct, there wouldn't be the possibility of accidentally leaving the extra `gboolean *out_enabled` uninitialized, etc.

File: src/libostree/ostree-bootloader-zipl.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (nikita-dubrovskii) [reply src/libostree/ostree-bootloader-zipl.c]: that's why GError was skipped at the begining )

**Pull request merged**